gateway/ln-gateway/src/lightning/mod.rs

```diff
@@ -248,6 +250,21 @@ impl dyn ILnRpcClient {
             })?;
         Ok((node_pub_key, alias, network, block_height, synced_to_chain))
     }
+
+    /// Waits for the Lightning node to be synced to the Bitcoin blockchain.
+    pub async fn wait_for_chain_sync(&self) -> std::result::Result<(), LightningRpcError> {
+        loop {
+            let info = self.info().await?;
+            let synced = info.synced_to_chain;
+            let block_height = info.block_height;
+            if synced {
+                return Ok(());
+            }
+
+            warn!(?block_height, "Lightning node is not synced");
+            fedimint_core::task::sleep(Duration::from_secs(30)).await;
```

Comment:
We'll be invoking this code from devimint, right? Won't this significantly slow down dev fed startup?

Reply:
*heart attack*

Reply:
@m1sterc001guy thoughts?

Reply:
Haha appreciate the patience :)

This is not currently called from devimint, it's only called from the gateway. Devimint has its own loop that is tighter than this one, so this does not cause devimint to be slower.

All that said, I think this should refactor devimint. With this change devimint shouldn't need to explicitly wait for chain sync since the gateway will just do it itself. I will make this change.